frontend/packages/dev-console/src/components/pipelines/pipeline-builder/utils.ts

```diff
@@ -98,7 +99,9 @@ export const convertBuilderFormToPipeline = (
     },
     spec: {
       ...existingPipeline?.spec,
-      params,
+      params: params.map((param) =>
+        _.isEmpty(param.default) ? { name: param.name, description: param.description } : param,
```

Comment:
@debsmita1 why do you have two different implementations for removing default when it is empty?